Free lnaddress Extension

bitkarrot opened the pull request:
As requested by @arcbtc :) 

As discussed here
https://github.com/lnbits/lnbits/issues/1035#issuecomment-1371846636
and 
https://github.com/bitkarrot/free-lnaddress-lnbits/issues/2

bitkarrot commented:
p.s. I tried to keep up with all the changes in the mainline but there are a bunch of things mypy didn't like, although files are nearly the same between this and lnurlp extension.....

bitkarrot commented:
There is a meta data bug on this branch I will try to get it fixed soon.

callebtc commented:
Some merge conflicts and mypy nags remain :) 🙏

callebtc commented on lnbits/core/views/public_api.py:
```diff
@@ -16,10 +17,17 @@
+        domain = urlparse(str(request.url)).netloc
+        return await lnurl_response(username, domain, request)
+
+    elif "lnaddy" not in settings.lnbits_disabled_extensions:
```
This means that it is either handled by `lndaddress` **or** `lnaddy`, is this intended?

bitkarrot replied:
There are two extensions that handle lightning addresses, there the one in this pull request (the free admin lnaddress) and then the cloudflare paid lightning extension. 

I was not sure how you want to address this conflict, as I originally wrote this as a personal extension, but right now as it stands to avoid name conflict on the local instance, the lnaddress extension has to be disabled in order for the lnaddy extension to work. 

If you feel there is a better way around this like checking both and allowing both extensions to be active, then we should do it.

bitkarrot replied:
I added a note in the .example.env and will add a note to the README.md in extension repo over here: https://github.com/lnbits/free-lnaddress-extension

bitkarrot commented:
> Some merge conflicts and mypy nags remain :) 🙏

I am working on fixing the mypy nags. Thanks, @callebtc

bitkarrot commented:
> Some merge conflicts and mypy nags remain :) 🙏

I fixed the mypy and formatting issues, but when i tried to pull in latest updates from main, there were a lot of issues with boltz extension so i reverted to pre-merge. It looks like now there are issues with lndwallet which i did not touch, so not sure what is going on. Halp? @callebtc

callebtc commented:
These errors were unrelated to your code, we have some issues with the CI pipeline that we're addressing though!

callebtc commented on lnbits/core/views/public_api.py:
```diff
@@ -38,7 +47,7 @@ async def api_public_payment_longpolling(payment_hash):
         if expiration < datetime.datetime.now():
             return {"status": "expired"}
-    except:
+    except HTTPException:
```
```suggestion
    except:
```

callebtc replied:
This should stay

dergigi commented:
I see that all checks passed for `72aeb4b` - any other issues, or could this be merged?

bitkarrot commented:
> There is a meta data bug on this branch I will try to get it fixed soon.
> 
> `Note: when trying to pay the invoice, I recieve an alert back Invoice description_hash doesn't match metadata.`
> 
> [bitkarrot/free-lnaddress-lnbits#4 (comment)](https://github.com/bitkarrot/free-lnaddress-lnbits/issues/4#issuecomment-1374442012)

This is now fixed.

bitkarrot commented:
Minor Conflicts from Main branch on .env.example and lnbits/extensions/lnaddress/lnurl.py fixed and merged. @callebtc

bitkarrot commented:
Hi @arcbtc,

As requested - The content in the extension folder has been moved to https://github.com/lnbits/free-lnaddress-extension

**Please note this pull request also includes changes to the lnbits core In order for extension to work!!**
